CSharpAdvanced/CSharpAdvanced/FlowerRepainter.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+namespace CSharpAdvanced
+{
+    static public class FlowerRepainter<T> where T : Flower
+    {
+        private static List<string> colors = new List<string>()
+        {
+            "Orange",
+            "Yellow",
+            "Green",
+            "Blue",
+            "Indigo",
+            "Violet"
+        };
+
+        public static void IncreasePrice(T flower)
+        {
+            flower.Price *= 1.15;
+        }
+
+        public static void RepaintFlower(T flower)
+        {
+            Console.WriteLine($"\nChoose the color for your {flower.Name}. Enter the number:");
```

Comment:
please add verification that if the color stays the same you will not increase price :)

Reply:
I have this verification on the 30 line:)